core/src/main/java/org/apache/calcite/rel/metadata/RelMdAllPredicates.java

```diff
@@ -120,12 +124,35 @@ public RelOptPredicateList getAllPredicates(Project project, RelMetadataQuery mq
   }
 
   /**
-   * Add the Filter condition to the list obtained from the input.
+   * Extract predicates for a Filter.
    */
   public RelOptPredicateList getAllPredicates(Filter filter, RelMetadataQuery mq) {
-    final RelNode input = filter.getInput();
-    final RexBuilder rexBuilder = filter.getCluster().getRexBuilder();
-    final RexNode pred = filter.getCondition();
+    return getAllFilterPredicates(filter, mq, filter.getCondition());
+  }
+
+  /**
+   * Extract predicates for a Calc.
+   */
+  public RelOptPredicateList getAllPredicates(Calc calc, RelMetadataQuery mq) {
+    final RexProgram rexProgram = calc.getProgram();
+    List<RexNode> condition = new ArrayList<>();
+    if (rexProgram.getCondition() != null) {
+      condition = Util.transform(ImmutableList.of(rexProgram.getCondition()),
+          rexProgram::expandLocalRef);
+    } else {
+      return mq.getAllPredicates(calc.getInput());
```

Comment:
The type of `condition` can be RexNode rather than List<RexNode>.

Reply:
Thanks, update the code.